### PR TITLE
Update README_LINUX.md (adds uninstall target)

### LINKED ISSUE #6730: Error when calling 'make uninstall'
I'm currently switching branches as I have an open pull request for 3.14 but normally I'm using 3.13.1. So, usually, after switching to another branch, I'm recompiling SuperCollider and then call
```
sudo make uninstall && sudo make install
```
However, there appears to be a problem with `cmake_uninstall.cmake` in 3.14:
```
CMake Error at cmake_uninstall.cmake:14 (message):
  Syntax error in cmake code at

    /home/stefan/Sources/SuperCollider/build/cmake_uninstall.cmake:14

  when parsing string

    Uninstalling \"${ENV{DESTDIR}}${file}\"

  Invalid character ('{') in a variable name: 'ENV'
```
I've been able to fix this by  adding the following block right before the `foreach` loop in `cmake_modules/cmake_uninstall.cmake.in`:
```cmake
if(DEFINED "${ENV}")
  set(ENV_DESTDIR "${ENV{DESTDIR}}")
else()
  set(ENV_DESTDIR "")
endif()
```
... and replacing every subsequent occurrence of `${ENV{DESTDIR}}` with `${ENV_DESTDIR}`.

I don't know if that's a proper fix (`ENV` doesn't seem to be defined on my machine, so I can't say what happens if it is) but if it is I could open a pull request for it.

## Environment

* SuperCollider version: 3.14-develop
* Operating system: Arch Linux
* Other details (Qt version, audio driver, etc.): CMake 4

## Steps to reproduce

```bash
sudo make uninstall
```

## Expected vs. actual behavior

Should uninstall all files of the old SuperCollider installation. Instead I get the error message posted above.

### PR DESCRIPTION
**Pull request by prko**
closes: #6730

## Purpose and Motivation
On Linux, SuperCollider installations created via `sudo make install` or `make install` cannot be removed using `sudo make uninstall` or `make uninstall`, because the build system does not provide an uninstall target. Users must instead remove the installed files manually. This PR adds documentation describing the required steps.

The procedure was suggested by Microsoft Copilot, and I have verified that it works. However, as I am not a daily Linux user, there may still be edge cases I might have overlooked.

## Types of changes

- Documentation

## To-do list

- [ ] Code is tested
- [ ] All tests are passing
- [x] Updated documentation
- [x] This PR is ready for review

### CODE REVIEW

**Comment by JordanHendersonMusic** [comment]
We should just add an uninstall target instead.

https://gitlab.kitware.com/cmake/community/-/wikis/FAQ#can-i-do-make-uninstall-with-cmake

**Comment by prko** [comment]
Thanks!, I hope this is correct...

File: cmake_uninstall.cmake.in [hunk 1]
```diff
@@ -7,10 +7,10 @@ string(REGEX REPLACE "\n" ";" files "${files}")
 foreach(file ${files})
   message(STATUS "Uninstalling $ENV{DESTDIR}${file}")
   if(IS_SYMLINK "$ENV{DESTDIR}${file}" OR EXISTS "$ENV{DESTDIR}${file}")
-    exec_program(
-      "@CMAKE_COMMAND@" ARGS "-E remove \"$ENV{DESTDIR}${file}\""
+    execute_process(
+      COMMAND "@CMAKE_COMMAND@" -E remove "$ENV{DESTDIR}${file}"
```
Comment (JordanHendersonMusic) [inline cmake_uninstall.cmake.in:11]: I've never seen '@' used like this... what does it do, and why not use `${CMAKE_COMMAND}" ?

Comment (prko) [reply cmake_uninstall.cmake.in:11]: I copied the code from the documentation you mentioned, so I wasn’t fully aware of the details.  
According to an AI explanation, `cmake_uninstall.cmake.in` is a template file, and `@CMAKE_COMMAND@` is replaced with the actual path during the configure step.

Comment (JordanHendersonMusic) [reply cmake_uninstall.cmake.in:11]: Ah yes! It is called in configure file, okay that makes sense.

**Review by capital-G — APPROVED** [review]
lgtm, uses the cmake example verbatim. thx prko and jordan!

**Comment by JordanHendersonMusic** [comment]
Ah, I think this might have been merged a little too quick...

There is already an uninstall cmake script.

https://github.com/supercollider/supercollider/blob/develop/cmake_modules/cmake_uninstall.cmake.in

I'd say this file should also go in the cmake_modules.

Given this file already exists, are we sure we don't already have an uninstall script somewhere?

**Comment by prko** [comment]
Here are the search results from the files in the SuperCollider repository:
<summary>cmake_uninstall.cmake.in: 4 results - 4 files</summary>
<details>
<pre>
CMakeLists.txt:
  46                        ${CMAKE_MODULE_PATH})
  47  
  48  CONFIGURE_FILE(
  49:   "${CMAKE_CURRENT_SOURCE_DIR}/cmake_modules/cmake_uninstall.cmake.in"
  50    "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake"
  51    IMMEDIATE @ONLY)
  52  

editors/scvim/CMakeLists.txt:
  10  
  11  IF(NOT TARGET uninstall)
  12    configure_file(
  13:     "${CMAKE_CURRENT_SOURCE_DIR}/cmake_uninstall.cmake.in"
  14      "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake"
  15      IMMEDIATE @ONLY)
  16  

external_libraries/portaudio/portaudio_submodule/CMakeLists.txt:
  466  
  467    IF (NOT TARGET uninstall)
  468      CONFIGURE_FILE(
  469:       "${CMAKE_CURRENT_SOURCE_DIR}/cmake_support/cmake_uninstall.cmake.in"
  470        "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake"
  471        IMMEDIATE @ONLY)
  472      ADD_CUSTOM_TARGET(uninstall

external_libraries/yaml-cpp/CMakeLists.txt:
  204  # uninstall target
  205  if(YAML_CPP_INSTALL AND NOT YAML_CPP_DISABLE_UNINSTALL AND NOT TARGET uninstall)
  206    configure_file(
  207:     "${CMAKE_CURRENT_SOURCE_DIR}/cmake_uninstall.cmake.in"
  208      "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake"
  209      IMMEDIATE @ONLY)
  210  
</pre>
</details>

<summary>uninstall: 100 results - 26 files</summary>
<details>
<pre>
CHANGELOG_archive.md:
  1650  - plugins: disable simd-optimization for tanh
  1651  
  1652  James Harkins (3):
  1653: - Explicitly show the command to uninstall (for scons idiots like me).
  1654  - (3.4) PathName now sets tmp directory using Platform
  1655  - SimpleController:update would throw error if no actions had been 'put' in
  1656  

  2870            plugins: disable simd-optimization for tanh
  2871  
  2872      James Harkins (3):
  2873:           Explicitly show the command to uninstall (for scons idiots like me).
  2874            (3.4) PathName now sets tmp directory using Platform
  2875            SimpleController:update would throw error if no actions had been 'put' in
  2876  

CMakeLists.txt:
  50    "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake"
  51    IMMEDIATE @ONLY)
  52  
  53: ADD_CUSTOM_TARGET(uninstall
  54    "${CMAKE_COMMAND}" -P "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake")
  55: set_property(TARGET uninstall PROPERTY FOLDER Util)
  56  
  57  # identify target architecture
  58  if(WIN32)

README_LINUX.md:
  369  sudo ldconfig
  370  ```
  371  
  372: To uninstall:
  373  
  374  ```shell
  375: sudo make uninstall
  376  ```
  377  
  378: (or `make uninstall` if you did user-wide installation).
  379  
  380  ### Building a Debian package
  381  

README_WINDOWS.md:
   101  
   102  For SC itself just run the downloaded installer. If necessary select an
   103  alternative install path when prompted during installation. In order to
   104: uninstall, use the default procedure on Windows (Control panel -> Uninstall a
   105  program) or run Uninstall.exe from within the SC installation folder. Note
   106  that files created in the 'user support directory' are not deleted (see below
   107  for details and an explanation).

   435  an installer. You can copy the main SuperCollider folder created during install
   436  to another machine. The binary installer just packages this folder and adds a
   437  few installation details like registering the uninstaller in the system and
   438: removing some registry settings during uninstall. The userAppSupportDirectory
   439  will be created from scratch if SuperCollider doesn't find it during start-up.
   440  Whether you copy the SC install folder, or use the full installer, you will
   441  always loose the local customizations and additions stored in the

  1029    required libraries to run SC without depending on shared libraries other
  1030    than the system libraries guaranteed to be available on every Windows system.
  1031    The binary installer adds some comfort, enhances SC system integration and
  1032:   cleanup on uninstall, but it is not required to run SC autonomously.
  1033  
  1034  * Enable compiler optimizations for your local system (could cause a faster
  1035    binary, but could also cause build problems. Don't use this if building for

.git/logs/HEAD:
  158  c556ee3b11171d91478e4641c1fdc6f0393e9529 aaa3bf2de8695e643a9446d9f0557c428346ee72 prko <pyoungryang.ko@googlemail.com> 1784169132 +0900	commit: Update help_browser.cpp
  159  aaa3bf2de8695e643a9446d9f0557c428346ee72 cade795364b1e0cc1993072ea4237dfec6ce2b6d prko <pyoungryang.ko@googlemail.com> 1784184903 +0900	commit: Update help_browser.cpp
  160  cade795364b1e0cc1993072ea4237dfec6ce2b6d df283194fc78355ecbf509ead8557dc3c62e09b5 prko <pyoungryang.ko@googlemail.com> 1784205782 +0900	checkout: moving from topic/helpbrowser_key_shortcuts_supp to topic/linux_uninstall_info
  161: df283194fc78355ecbf509ead8557dc3c62e09b5 ad63c55f07018cded845a2a6f3943f2f0f50ff34 prko <pyoungryang.ko@googlemail.com> 1784206067 +0900	commit: adds uninstall target
  162  ad63c55f07018cded845a2a6f3943f2f0f50ff34 e747b1a728e7121c4e9c0345226b9719cf4bb3a0 prko <pyoungryang.ko@googlemail.com> 1784206087 +0900	commit: restore README_LINUX.md
  163  e747b1a728e7121c4e9c0345226b9719cf4bb3a0 de6327a5527f9ecd9ae61300e45d7af0b6343155 prko <pyoungryang.ko@googlemail.com> 1784208017 +0900	commit: fix errors
  164  de6327a5527f9ecd9ae61300e45d7af0b6343155 cade795364b1e0cc1993072ea4237dfec6ce2b6d prko <pyoungryang.ko@googlemail.com> 1784227298 +0900	checkout: moving from topic/linux_uninstall_info to topic/helpbrowser_key_shortcuts_supp

.git/logs/refs/heads/topic/linux_uninstall_info:
  1  0000000000000000000000000000000000000000 520821479c66811d0d131251e77e12d6fe07921f prko <pyoungryang.ko@googlemail.com> 1784152869 +0900	branch: Created from develop
  2  520821479c66811d0d131251e77e12d6fe07921f df283194fc78355ecbf509ead8557dc3c62e09b5 prko <pyoungryang.ko@googlemail.com> 1784153141 +0900	commit: Update README_LINUX.md
  3: df283194fc78355ecbf509ead8557dc3c62e09b5 ad63c55f07018cded845a2a6f3943f2f0f50ff34 prko <pyoungryang.ko@googlemail.com> 1784206067 +0900	commit: adds uninstall target
  4  ad63c55f07018cded845a2a6f3943f2f0f50ff34 e747b1a728e7121c4e9c0345226b9719cf4bb3a0 prko <pyoungryang.ko@googlemail.com> 1784206087 +0900	commit: restore README_LINUX.md
  5  e747b1a728e7121c4e9c0345226b9719cf4bb3a0 de6327a5527f9ecd9ae61300e45d7af0b6343155 prko <pyoungryang.ko@googlemail.com> 1784208017 +0900	commit: fix errors

.github/workflows/build_macos.yml:
   21              deployment-target: "11"
   22              cmake-architectures: "x86_64;arm64"
   23              homebrew-packages: ""
   24:             homebrew-uninstall: "readline"
   25              vcpkg-packages: "readline libsndfile"
   26              vcpkg-triplet: "arm64-osx-release-supercollider"
   27              vcpkg-triplet-other: "x64-osx-release-supercollider"

   40              homebrew-packages: "readline portaudio"
   41              vcpkg-packages: "libsndfile"
   42              # homebrew-packages: "" # use this instead when cross-compiling for x86_64 on arm64
   43:             # homebrew-uninstall: "readline" # use this instead when cross-compiling for x86_64 on arm64
   44              # vcpkg-packages: "readline portaudio libsndfile" # use this instead when cross-compiling for x86_64 on arm64
   45              vcpkg-triplet: x64-osx-release-supercollider # required for build-libsndfile
   46              extra-cmake-flags: "-D SC_COMPILER_ARCH_FLAGS='-march=ivybridge'"

  147          if: matrix.homebrew-packages
  148          run: brew install ${{ matrix.homebrew-packages }} --quiet
  149  
  150:       - name: uninstall homebrew packages
  151:         if: matrix.homebrew-uninstall
  152:         run: brew uninstall --ignore-dependencies ${{ matrix.homebrew-uninstall }}
  153  
  154        - name: install vcpkg packages
  155          if: matrix.vcpkg-packages && matrix.vcpkg-triplet

editors/scvim/CMakeLists.txt:
   8      PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
   9  ENDIF()
  10  
  11: IF(NOT TARGET uninstall)
  12    configure_file(
  13      "${CMAKE_CURRENT_SOURCE_DIR}/cmake_uninstall.cmake.in"
  14      "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake"
  15      IMMEDIATE @ONLY)
  16  
  17:   add_custom_target(uninstall
  18      COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake)
  19: ENDIF(NOT TARGET uninstall)

external_libraries/portaudio/portaudio_submodule/CMakeLists.txt:
  437  # The option below offers the option to avoid executing the portaudio install target
  438  # for cases in which the host-application executes install, but no independent install
  439  # of portaudio is wished.
  440: OPTION(PA_DISABLE_INSTALL "Disable targets install and uninstall (for embedded builds)" OFF)
  441  
  442  IF(NOT PA_OUTPUT_OSX_FRAMEWORK AND NOT PA_DISABLE_INSTALL)
  443    INCLUDE(CMakePackageConfigHelpers)

  464                  "${CMAKE_BINARY_DIR}/cmake/portaudio/portaudioConfigVersion.cmake"
  465      DESTINATION "lib/cmake/portaudio")
  466  
  467:   IF (NOT TARGET uninstall)
  468      CONFIGURE_FILE(
  469        "${CMAKE_CURRENT_SOURCE_DIR}/cmake_support/cmake_uninstall.cmake.in"
  470        "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake"
  471        IMMEDIATE @ONLY)
  472:     ADD_CUSTOM_TARGET(uninstall
  473        COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake)
  474    ENDIF()
  475  ENDIF()

external_libraries/portaudio/portaudio_submodule/ltmain.sh:
    56  #         finish             complete the installation of libtool libraries
    57  #         install            install libraries or executables
    58  #         link               create a library or an executable
    59: #         uninstall          remove libraries from an installed directory
    60  #
    61  # MODE-ARGS vary depending on the MODE.  When passed as first option,
    62  # `--mode=MODE' may be abbreviated as `MODE' or a unique abbreviation of that.

  1031  link|lin|li|l)
  1032    shift; set dummy --mode link ${1+"$@"}; shift
  1033    ;;
  1034: uninstall|uninstal|uninsta|uninst|unins|unin|uni|un|u)
  1035:   shift; set dummy --mode uninstall ${1+"$@"}; shift
  1036    ;;
  1037  esac
  1038  

  1103  			opt_mode="$optarg"
  1104  case $optarg in
  1105    # Valid mode arguments:
  1106:   clean|compile|execute|finish|install|link|relink|uninstall) ;;
  1107  
  1108    # Catch anything else as an error
  1109    *) func_error "invalid argument for $opt"

  2458  is created, otherwise an executable program is created."
  2459          ;;
  2460  
  2461:       uninstall)
  2462          $ECHO \
  2463: "Usage: $progname [OPTION]... --mode=uninstall RM [RM-OPTION]... FILE...
  2464  
  2465  Remove libraries from an installation directory.
  2466  

  2488    else
  2489      {
  2490        func_help noexit
  2491:       for opt_mode in compile link execute install finish uninstall clean; do
  2492  	func_mode_help
  2493        done
  2494      } | sed -n '1p; 2,$s/^Usage:/  or: /p'
  2495      {
  2496        func_help noexit
  2497:       for opt_mode in compile link execute install finish uninstall clean; do
  2498  	echo
  2499  	func_mode_help
  2500        done

  9481        fi
  9482        func_basename "$file"
  9483        name="$func_basename_result"
  9484:       test "$opt_mode" = uninstall && odir="$dir"
  9485  
  9486        # Remember odir for removal later, being careful to avoid duplicates
  9487        if test "$opt_mode" = clean; then

  9525  	    esac
  9526  	    test -n "$libdir" && func_append rmfiles " $odir/$name $odir/${name}i"
  9527  	    ;;
  9528: 	  uninstall)
  9529  	    if test -n "$library_names"; then
  9530  	      # Do each command in the postuninstall commands.
  9531  	      func_execute_cmds "$postuninstall_cmds" 'test "$rmforce" = yes || exit_status=1'

  9614      exit $exit_status
  9615  }
  9616  
  9617: { test "$opt_mode" = uninstall || test "$opt_mode" = clean; } &&
  9618      func_mode_uninstall ${1+"$@"}
  9619  
  9620  test -z "$opt_mode" && {

external_libraries/portaudio/portaudio_submodule/Makefile.in:
  208  	@echo ""
  209  	$(MAKE) install-recursive
  210  
  211: uninstall:
  212: 	$(LIBTOOL) --mode=uninstall rm -f $(DESTDIR)$(libdir)/$(PALIB)
  213: 	$(LIBTOOL) --mode=uninstall rm -f $(DESTDIR)$(includedir)/portaudio.h
  214: 	$(MAKE) uninstall-recursive
  215  
  216  clean:
  217  	$(LIBTOOL) --mode=clean rm -f $(LTOBJS) $(LOOPBACK_OBJS) $(ALL_TESTS) lib/$(PALIB)

  254  install-recursive:
  255  	if test -n "$(SUBDIRS)" ; then for dir in "$(SUBDIRS)"; do $(MAKE) -C $$dir install; done ; fi
  256  
  257: uninstall-recursive:
  258: 	if test -n "$(SUBDIRS)" ; then for dir in "$(SUBDIRS)"; do $(MAKE) -C $$dir uninstall; done ; fi

external_libraries/portaudio/portaudio_submodule/bindings/cpp/INSTALL:
  83       all sorts of other programs in order to regenerate files that came
  84       with the distribution.
  85  
  86:   7. Often, you can also type `make uninstall' to remove the installed
  87       files again.  In practice, not all packages have tested that
  88       uninstallation works correctly, even though it is required by the
  89       GNU Coding Standards.
  90  
  91    8. Some packages, particularly those that use Automake, provide `make
  92       distcheck', which can by used by developers to test that all other
  93:      targets like `make install' and `make uninstall' work correctly.
  94       This target is generally not run by end users.
  95  
  96  Compilers and Options

external_libraries/portaudio/portaudio_submodule/bindings/cpp/Makefile.in:
  116  	install-info-recursive install-pdf-recursive \
  117  	install-ps-recursive install-recursive installcheck-recursive \
  118  	installdirs-recursive pdf-recursive ps-recursive \
  119: 	tags-recursive uninstall-recursive
  120  am__can_run_installinfo = \
  121    case $$AM_UPDATE_INFO_DIR in \
  122      n|no|NO) false;; \

  415  	  $(INSTALL_DATA) $$files "$(DESTDIR)$(pkgconfigdir)" || exit $$?; \
  416  	done
  417  
  418: uninstall-pkgconfigDATA:
  419  	@$(NORMAL_UNINSTALL)
  420  	@list='$(pkgconfig_DATA)'; test -n "$(pkgconfigdir)" || list=; \
  421  	files=`for p in $$list; do echo $$p; done | sed -e 's|^.*/||'`; \

  668  	  && $(MAKE) $(AM_MAKEFLAGS) check \
  669  	  && $(MAKE) $(AM_MAKEFLAGS) install \
  670  	  && $(MAKE) $(AM_MAKEFLAGS) installcheck \
  671: 	  && $(MAKE) $(AM_MAKEFLAGS) uninstall \
  672  	  && $(MAKE) $(AM_MAKEFLAGS) distuninstallcheck_dir="$$dc_install_base" \
  673  	        distuninstallcheck \
  674  	  && chmod -R a-w "$$dc_install_base" \
  675  	  && ({ \
  676  	       (cd ../.. && umask 077 && mkdir "$$dc_destdir") \
  677  	       && $(MAKE) $(AM_MAKEFLAGS) DESTDIR="$$dc_destdir" install \
  678: 	       && $(MAKE) $(AM_MAKEFLAGS) DESTDIR="$$dc_destdir" uninstall \
  679  	       && $(MAKE) $(AM_MAKEFLAGS) DESTDIR="$$dc_destdir" \
  680  	            distuninstallcheck_dir="$$dc_destdir" distuninstallcheck; \
  681  	      } || { rm -rf "$$dc_destdir"; exit 1; }) \

  700  	  exit 1; \
  701  	}; \
  702  	test `$(am__distuninstallcheck_listfiles) | wc -l` -eq 0 \
  703: 	   || { echo "ERROR: files left after uninstall:" ; \
  704  	        if test -n "$(DESTDIR)"; then \
  705  	          echo "  (check DESTDIR support)"; \
  706  	        fi ; \

  726  install: install-recursive
  727  install-exec: install-exec-recursive
  728  install-data: install-data-recursive
  729: uninstall: uninstall-recursive
  730  
  731  install-am: all-am
  732  	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am

  821  
  822  ps-am:
  823  
  824: uninstall-am: uninstall-pkgconfigDATA
  825  
  826  .MAKE: $(am__recursive_targets) install-am install-strip
  827  

  839  	install-ps-am install-strip installcheck installcheck-am \
  840  	installdirs installdirs-am maintainer-clean \
  841  	maintainer-clean-generic mostlyclean mostlyclean-generic \
  842: 	mostlyclean-libtool pdf pdf-am ps ps-am tags tags-am uninstall \
  843: 	uninstall-am uninstall-pkgconfigDATA
  844  
  845  
  846  # Tell versions [3.59,3.63) of GNU make to not export all variables.

external_libraries/portaudio/portaudio_submodule/bindings/cpp/bin/Makefile.in:
  498  install: install-am
  499  install-exec: install-exec-am
  500  install-data: install-data-am
  501: uninstall: uninstall-am
  502  
  503  install-am: all-am
  504  	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am

  594  
  595  ps-am:
  596  
  597: uninstall-am:
  598  
  599  .MAKE: install-am install-strip
  600  

  610  	installcheck-am installdirs maintainer-clean \
  611  	maintainer-clean-generic mostlyclean mostlyclean-compile \
  612  	mostlyclean-generic mostlyclean-libtool pdf pdf-am ps ps-am \
  613: 	tags tags-am uninstall uninstall-am
  614  
  615  
  616  # Tell versions [3.59,3.63) of GNU make to not export all variables.

external_libraries/portaudio/portaudio_submodule/bindings/cpp/doc/Makefile.in:
  315  install: install-am
  316  install-exec: install-exec-am
  317  install-data: install-data-am
  318: uninstall: uninstall-am
  319  
  320  install-am: all-am
  321  	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am

  406  
  407  ps-am:
  408  
  409: uninstall-am:
  410  
  411  .MAKE: install-am install-strip
  412  

  420  	install-strip installcheck installcheck-am installdirs \
  421  	maintainer-clean maintainer-clean-generic mostlyclean \
  422  	mostlyclean-generic mostlyclean-libtool pdf pdf-am ps ps-am \
  423: 	tags-am uninstall uninstall-am
  424  
  425  #INCLUDES = -I$(srcdir)/$(PACPP_ROOT)/include -I$(top_srcdir)/include
  426  

external_libraries/portaudio/portaudio_submodule/bindings/cpp/include/Makefile.in:
  356  	  $(INSTALL_HEADER) $$files "$(DESTDIR)$(pkgincludedir)" || exit $$?; \
  357  	done
  358  
  359: uninstall-pkgincludeHEADERS:
  360  	@$(NORMAL_UNINSTALL)
  361  	@list='$(pkginclude_HEADERS)'; test -n "$(pkgincludedir)" || list=; \
  362  	files=`for p in $$list; do echo $$p; done | sed -e 's|^.*/||'`; \

  454  install: install-am
  455  install-exec: install-exec-am
  456  install-data: install-data-am
  457: uninstall: uninstall-am
  458  
  459  install-am: all-am
  460  	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am

  545  
  546  ps-am:
  547  
  548: uninstall-am: uninstall-pkgincludeHEADERS
  549  
  550  .MAKE: install-am install-strip
  551  

  560  	install-ps-am install-strip installcheck installcheck-am \
  561  	installdirs maintainer-clean maintainer-clean-generic \
  562  	mostlyclean mostlyclean-generic mostlyclean-libtool pdf pdf-am \
  563: 	ps ps-am tags tags-am uninstall uninstall-am \
  564: 	uninstall-pkgincludeHEADERS
  565  
  566  
  567  #       portaudiocpp/AsioDeviceAdapter.hxx

external_libraries/portaudio/portaudio_submodule/bindings/cpp/lib/Makefile.in:
  392  	  $(LIBTOOL) $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=install $(INSTALL) $(INSTALL_STRIP_FLAG) $$list2 "$(DESTDIR)$(libdir)"; \
  393  	}
  394  
  395: uninstall-libLTLIBRARIES:
  396  	@$(NORMAL_UNINSTALL)
  397  	@list='$(lib_LTLIBRARIES)'; test -n "$(libdir)" || list=; \
  398  	for p in $$list; do \
  399  	  $(am__strip_dir) \
  400: 	  echo " $(LIBTOOL) $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=uninstall rm -f '$(DESTDIR)$(libdir)/$$f'"; \
  401: 	  $(LIBTOOL) $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=uninstall rm -f "$(DESTDIR)$(libdir)/$$f"; \
  402  	done
  403  
  404  clean-libLTLIBRARIES:

  669  install: install-am
  670  install-exec: install-exec-am
  671  install-data: install-data-am
  672: uninstall: uninstall-am
  673  
  674  install-am: all-am
  675  	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am

  765  
  766  ps-am:
  767  
  768: uninstall-am: uninstall-libLTLIBRARIES
  769  
  770  .MAKE: install-am install-strip
  771  

  781  	installcheck installcheck-am installdirs maintainer-clean \
  782  	maintainer-clean-generic mostlyclean mostlyclean-compile \
  783  	mostlyclean-generic mostlyclean-libtool pdf pdf-am ps ps-am \
  784: 	tags tags-am uninstall uninstall-am uninstall-libLTLIBRARIES
  785  
  786  
  787  # Tell versions [3.59,3.63) of GNU make to not export all variables.

external_libraries/yaml-cpp/CMakeLists.txt:
  201      VERBATIM)
  202  endif()
  203  
  204: # uninstall target
  205: if(YAML_CPP_INSTALL AND NOT YAML_CPP_DISABLE_UNINSTALL AND NOT TARGET uninstall)
  206    configure_file(
  207      "${CMAKE_CURRENT_SOURCE_DIR}/cmake_uninstall.cmake.in"
  208      "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake"
  209      IMMEDIATE @ONLY)
  210  
  211:   add_custom_target(uninstall
  212      COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake)
  213  endif()

HelpSource/Classes/Quark.schelp:
   99  METHOD:: install
  100  returns:: this
  101  
  102: METHOD:: uninstall
  103  returns:: this
  104  
  105  METHOD:: checkout

HelpSource/Classes/Quarks.schelp:
  32  ARGUMENT:: quark
  33  returns:: this
  34  
  35: METHOD:: uninstall
  36  Will execute the link::Guides/UsingQuarks#Hooks#hooks:: code::\preUninstall:: and code::\postUninstall:: if defined.
  37  ARGUMENT:: name
  38  Name (String) of a quark that is listed in the directory,
  39  or url of a git repository
  40: or the path (absolute or relative to current working directory) of a folder to uninstall.
  41  returns:: this
  42  
  43  METHOD:: clear

HelpSource/Guides/UsingQuarks.schelp:
  54  
  55  code::
  56  Quarks.install("MathLib");
  57: Quarks.uninstall("MathLib");
  58  ::
  59  
  60  subsection::by git URL
  61  
  62  code::
  63  Quarks.install("https://github.com/supercollider-quarks/MathLib.git");
  64: // uninstall it
  65: Quarks.uninstall("https://github.com/supercollider-quarks/MathLib.git");
  66  ::
  67  
  68  subsection::from a local folder

  84  
  85  Now each quark inside that folder will appear on the GUI for you to un/install.
  86  
  87: If you wish to be able to uninstall local quarks, but still see them in the GUI, it is highly recommended to collect your local quark directories under a single primary directory. Then, use link::Classes/Quarks#*addFolder:: to add the primary directory. code::addFolder:: does not add individual quark directories.
  88  
  89  subsection::Recompile the class library
  90  

SCClassLibrary/Common/Collections/String.sc:
  449  	}
  450  	exclude {
  451  		if(Quarks.isInstalled(this)) {
  452: 			Quarks.uninstall(this);
  453  			"... the class library may have to be recompiled.".postln;
  454  		}
  455  	}

SCClassLibrary/Common/Quarks/Quark.sc:
  113  			("Quark" + name + "was not installed using git, cannot update.").warn;
  114  		});
  115  	}
  116: 	uninstall {
  117  		this.runHook(\preUninstall);
  118  		Quarks.uninstallQuark(this);
  119  		this.runHook(\postUninstall);

SCClassLibrary/Common/Quarks/Quarks.sc:
  34  			^quark
  35  		});
  36  	}
  37: 	*uninstall { |name|
  38  		// by quark name or local path
  39  		this.installed.do { |q|
  40  			if(q.name == name, {

SCClassLibrary/Common/Quarks/GUI/QuarksGui.sc:
  545  			quarksGui.setMsg("Installing" + model.name);
  546  			if(model.isInstalled, {
  547  				// reinstall possibly with different dependencies
  548: 				model.uninstall;
  549  			});
  550  			model = Quarks.install(model.url, refspec);
  551  		}.protect({ |err|

  597  				prev = Quarks.installed.detect({ |q| q.name == quark.name });
  598  				if(prev.notNil, {
  599  					"Uninstalling other version % %".format(prev, prev.localPath).warn;
  600: 					prev.uninstall;
  601  				});
  602  				ok = quark.install;
  603  				if(ok, {

  606  					quarksGui.setMsg("Error while installing" + quark, \error);
  607  				});
  608  			}, {
  609: 				quark.uninstall;
  610  				quarksGui.setMsg("Uninstalled" + quark, \success);
  611  			});
  612  		}, {

testsuite/classlibrary/TestQuark.sc:
  39  	test_runQuarkInstallHooks {
  40  		TestQuark.clearTempDir;
  41  		// as the quark to be installed is empty we can not update or
  42: 		// uninstall it.
  43  		// Making it non empty does not fix this as it is already
  44  		// in a subdir of the testsuite, therefore every class file
  45  		// will be already available and a proper Quark install
</pre>
</details>

File: CMakeLists.txt [hunk 2]
```diff
@@ -651,3 +651,15 @@ mark_as_advanced(MATH_LIBRARY)
 mark_as_advanced(QT_QMAKE_EXECUTABLE)
 mark_as_advanced(SNDFILE SNDFILE_LIBRARY_DIR)
 mark_as_advanced(SC_HIDAPI)
+
+#############################################
+# uninstall target
+if(NOT TARGET uninstall)
+  configure_file(
+    "${CMAKE_CURRENT_SOURCE_DIR}/cmake_uninstall.cmake.in"
```
Comment (capital-G) [inline CMakeLists.txt:659]: probably the existing file would have probably been fine, so

```suggestion
    "${CMAKE_SOURCE_DIR}/cmake_modules/cmake_uninstall.cmake.in"
```

**Review by capital-G — COMMENTED** [review]
Ah, sorry - jordan is right, a bit too eager execution...

I think we could somehow revert this through another PR which deletes the added file from root and uses the existing one.
One should perform a diff if it actually matches the file from the CMake FAQ.

File: CMakeLists.txt (hunk 2 above)
Comment (prko) [reply CMakeLists.txt:659]: Thanks! See #7646.
